bundles/org.openhab.binding.enturno/src/main/java/org/openhab/binding/enturno/internal/EnturNoHandler.java

```diff
@@ -0,0 +1,305 @@
+                    state = estimatedFlags.size() > 3 ? getStringTypeState(estimatedFlags.get(3)) : state;
+                    break;
+                case EnturNoBindingConstants.ESTIMATED_FLAG_05:
+                    state = estimatedFlags.size() > 4 ? getStringTypeState(estimatedFlags.get(4)) : state;
+                    break;
+                case EnturNoBindingConstants.CHANNEL_LINE_CODE:
+                    state = getStringTypeState(processedData.get(i).lineCode);
+                    break;
+                case EnturNoBindingConstants.CHANNEL_FRONT_DISPLAY:
+                    state = getStringTypeState(processedData.get(i).frontText);
+                    break;
+                default:
+                    break;
+            }
+            logger.debug("Update channel '{}' of group '{}' with new state '{}'.", channelId, channelGroupId, state);
+            updateState(channelUID, state);
+        } else {
+            logger.debug("No real-time data available to update channel '{}' of group '{}'.", channelId,
+                    channelGroupId);
+        }
+    }
+
+    /**
+     * Update the channel from the last Entur data retrieved.
+     *
+     * @param channelUID the id identifying the channel to be updated
+     */
+    private void updateStopPlaceChannel(ChannelUID channelUID) {
+        String channelId = channelUID.getIdWithoutGroup();
+        String channelGroupId = channelUID.getGroupId();
+        if (processedData.size() > 0) {
+            State state = UnDefType.UNDEF;
+            switch (channelId) {
+                case EnturNoBindingConstants.CHANNEL_STOP_ID:
+                    state = getStringTypeState(processedData.get(0).stopPlaceId);
+                    break;
+                case EnturNoBindingConstants.CHANNEL_STOP_NAME:
+                    state = getStringTypeState(processedData.get(0).stopName);
+                    break;
+                case EnturNoBindingConstants.CHANNEL_STOP_TRANSPORT_MODE:
+                    state = getStringTypeState(processedData.get(0).transportMode);
+                    break;
+                default:
+                    break;
+            }
+            logger.debug("Update channel '{}' of group '{}' with new state '{}'.", channelId, channelGroupId, state);
+            updateState(channelUID, state);
+        } else {
+            logger.debug("No real-time data available to update channel '{}' of group '{}'.", channelId,
+                    channelGroupId);
+        }
+    }
+
+    private State getDateTimeTypeState(@Nullable String value) {
+        return (value == null) ? UnDefType.UNDEF
+                : new DateTimeType(ZonedDateTime.parse(value, DateTimeFormatter.ISO_DATE_TIME)
+                        .withZoneSameInstant(ZoneId.of(EnturNoBindingConstants.TIME_ZONE)));
+    }
+
+    private State getStringTypeState(@Nullable String value) {
+        return (value == null) ? UnDefType.UNDEF : new StringType(value);
+    }
+}
```

Comment:
Please check for missing newlines

Reply:
Do you mean: 

```suggestion
return (value == null || !StringUtils.endsWith(value, "\n")) ? UnDefType.UNDEF : new StringType(value);
```

or 

```suggestion
return (value == null || !StringUtils.endsWith(value, System.lineSeparator())) ? UnDefType.UNDEF : new StringType(value);
```

Reply:
I mean literally missing new lines of files, it is part of our coding guide that every file has to end with a new line ^^

Reply:
Ok, I see :) I have also read about problems with parsing values from .items file that were missing new line character :) But this is fixed now